File: src/main/java/com/teamwizardry/wizardry/common/utils/ColorUtils.java
package com.teamwizardry.wizardry.common.utils;


import java.awt.*;
import java.util.List;

public class ColorUtils {

    public static Color generateRandomColor() {
        return new Color(RandUtil.nextFloat(), RandUtil.nextFloat(), RandUtil.nextFloat());
    }

    public static Color mergeColors(Color color1, Color color2) {
        return new Color((int) (color1.getRed() * 0.5 + color2.getRed() * 0.5),
                (int) (color1.getGreen() * 0.5 + color2.getGreen() * 0.5),
                (int) (color1.getBlue() * 0.5 + color2.getBlue() * 0.5));
    }

    public static Color[] mergeColorSets(List<Color[]> colors) {
        Color[] resultSet = null;

        for (Color[] colorSet : colors) {
            if (resultSet == null) {
                resultSet = colorSet;
                continue;
            }
            for (int i = 0, resultSetLength = resultSet.length; i < resultSetLength; i++) {
                Color resultColor = resultSet[i];
                resultSet[i] = mergeColors(resultColor, colorSet[i]);
            }
        }

        return resultSet;
    }
}

File: src/main/kotlin/com.teamwizardry.wizardry/common/utils/ColorUtils.kt
package com.teamwizardry.wizardry.common.utils

import java.awt.Color

object ColorUtils {
    fun generateRandomColor(): Color {
        return Color(RandUtil.nextFloat(), RandUtil.nextFloat(), RandUtil.nextFloat())
    }

    private fun mergeColors(color1: Color, color2: Color): Color {
        return Color(
            (color1.red * 0.5 + color2.red * 0.5).toInt(),
            (color1.green * 0.5 + color2.green * 0.5).toInt(),
            (color1.blue * 0.5 + color2.blue * 0.5).toInt()
        )
    }

    fun mergeColorSets(colors: List<Array<Color>>): Array<Color>? {
        var resultSet: Array<Color>? = null
        for (colorSet in colors) {
            if (resultSet == null) {
                resultSet = colorSet
                continue
            }
            var i = 0
            val resultSetLength = resultSet.size
            while (i < resultSetLength) {
                val resultColor = resultSet[i]
                resultSet[i] = mergeColors(resultColor, colorSet[i])
                i++
            }
        }
        return resultSet
    }
}